io/src/lib.rs

```diff
@@ -350,56 +352,3 @@ pub mod io {
     /// Returns a sink to which all writes succeed. See [`std::io::sink`] for more info.
     pub fn sink() -> Sink { Sink }
 }
-
-#[doc(hidden)]
-#[cfg(feature = "std")]
-/// Re-export std for the below macro
-pub use std as _std;
```

Comment:
I think the reason crates do this is because other crates can rename a crate to be named e.g. `std`. But honestly, if anyone is using this crate with `std` feature and then instead of using an actual `std` renames a crate to `std` that also has incompatible API they deserve all the breakage they'd get.